loom/graphics/gfxTexture.cpp

```diff
@@ -417,26 +530,26 @@ void Texture::dispose(TextureID id)
         return;
     }
 
+    loom_mutex_lock(Texture::sTexInfoLock);
     TextureInfo *tinfo = &sTextureInfos[id];
 
     // If the texture isn't valid ignore it.
-    if (tinfo->handle.idx == bgfx::invalidHandle)
+    if (tinfo->handle.idx != bgfx::invalidHandle)
     {
-        return;
-    }
-
-    //TODO: LOOM-1653, we really shouldn't be holding a copy of the texture data in the
-    // asset system until we dispose
-    if (!tinfo->texturePath.empty()) {
-        loom_asset_unsubscribe(tinfo->texturePath.c_str(), handleAssetNotification, (void *)id);
-        loom_asset_flush(tinfo->texturePath.c_str());
+        //TODO: LOOM-1653, we really shouldn't be holding a copy of the texture data in the
+        // asset system until we dispose
+        if (!tinfo->texturePath.empty()) {
+            loom_asset_unsubscribe(tinfo->texturePath.c_str(), handleAssetNotification, (void *)id);
+            loom_asset_flush(tinfo->texturePath.c_str());
+
+            // Reset the hash, too
+            sTexturePathLookup.erase(tinfo->texturePath);
+        }
 
-        // Reset the hash, too
-        sTexturePathLookup.erase(tinfo->texturePath);
+        // And erase backing state.
+        bgfx::destroyTexture(tinfo->handle);
```

Comment:
I'm not sure; that destroy code has been around forever... as far as I understand, there is no path where you could call Texture::dispose() from any thread other than the main thread (unless you started up your own thread to do that...).  In other words, I didn't add anything that changed how/where Dispose is called from, so if it was going to break, it likely would have broken long before, no?